pygdf/_gdf.py

```diff
@@ -311,3 +311,44 @@ def hash_columns(columns, result):
     hashfn = libgdf.GDF_HASH_MURMUR3
     libgdf.gdf_hash(ncols, col_input, hashfn, col_out)
     return result
+
+
+def hash_partition(input_columns, key_indices, nparts, output_columns):
+    """Partition the input_columns by the hash values on the keys.
+
+    Parameters
+    ----------
+    input_columns : sequence of Column
+    key_indices : sequence of int
+        Indices into `input_columns` that indicates the key columns.
+    nparts : int
+        number of partitions
+
+    Returns
+    -------
+    partition_offsets : list of int
+        Each index indicates the start of a partition.
+    """
+    assert len(input_columns) == len(output_columns)
+    for col in input_columns:
+        if col.null_count != 0:
+            raise ValueError('cannot handle masked column')
+
+    col_inputs = [col.cffi_view for col in input_columns]
+    col_outputs = [col.cffi_view for col in output_columns]
+    offsets = ffi.new('int[]', nparts)
+    hashfn = libgdf.GDF_HASH_MURMUR3
```

Comment:
Are there plans to support other algorithms soon? In that case it might be better to have this separate similar to `np_to_gdf_dtype(dtype)` and have it as a parameter to `hash_partition` like `hash_type='murmur3`

Reply:
It's completely depending on the libgdf library.  There's just one hashfn right now in https://github.com/gpuopenanalytics/libgdf/blob/cf65f3490428183355e47a90620e951981327d4a/include/gdf/cffi/types.h#L45-L47.    I think we can wait til a new hashfn are available  before we make it configurable.